File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <map>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, int> count;

        for (const int& num : nums){
            count[num] += 1;
        }

        for (auto i : count) {
            if (i.second > 1) {
                return true;
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = defaultdict(int)

        for num in nums:
            freqs[num] += 1

        sorted_freq = sorted(freqs.items(), key=lambda item: item[1])

        val = []

        for freq in sorted_freq:
            val.append(freq[0])

        return val[-k::]